Applications/Spire/Include/Spire/SpireTester/TestOrderImbalanceIndicatorModel.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef SPIRE_TEST_ORDER_IMBALANCE_INDICATOR_MODEL_HPP
+#define SPIRE_TEST_ORDER_IMBALANCE_INDICATOR_MODEL_HPP
+#include <Beam/Threading/ConditionVariable.hpp>
+#include <Beam/Threading/Mutex.hpp>
+#include "Spire/OrderImbalanceIndicator/OrderImbalanceIndicatorModel.hpp"
+
+namespace Spire {
+
+  //! Represents an OrderImbalanceIndicatorModel for testing, allowing for the
+  //! observation of subscription requests sent to the model and simulation
+  //! of asynchronous loads.
+  class TestOrderImbalanceIndicatorModel :
+      public OrderImbalanceIndicatorModel {
+    public:
+
+      //! Stores a request to subscribe to an OrderImbalanceModel.
+      class SubscribeEntry {
```

Comment:
Generally classes and "data structures" are named after nouns since they represent objects. Functions are named as verbs since they perform an action. If you wish to use "subscribe" in its noun form to describe the "entry" then you need to rename this to ```SubscriptionEntry``` since subscription is a noun whereas subscribe is a verb.

You named the function ```subscribe``` which is correct, but you incorrectly named some other functions like ```pop_subscribe``` which should be ```pop_subscription``` since "pop" is the verb and "subscription" is the noun. Go over this class and revise the names to use the noun or verb depending on the situation.